File: challenges/js/strings/super_reduced_string.js
function processData(input) {
    //Enter your code here
    result = "";
    for(let i = 0; i < input.length; i++){
        if(input[i] !== input[i + 1]){
            result += input[i];
        } else{
            i++;
        }
    }
    if(result === ""){
        console.log("Empty String");
    } else{
        console.log(result);
    }

}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: challenges/ruby/hackerrank/strings/super_reduced_string.rb
# Enter your code here. Read input from STDIN. Print output to STDOUT
input = gets.strip

i = 0
finished = false

while !finished
    finished = true
    result = ""
    i = 0

    while i < input.length
        if input[i] == input[i + 1]
            i += 1
            finished = false
        else
            result += input[i]
        end
        i += 1
    end
    input = result
end

puts input 
